tests/testthat/test-fitPoisBG.R

```diff
@@ -14,34 +14,56 @@ res <- fitPoisBG(demoData, size_scale = "first")
 # spec: "The function outputs a GeoMx S4 class with length same as length of ROIs, sizefact, in phenoData.
 #       The value is NA for non-negative probes."
 test_that("fitPoisBG returns a well-formatted geomxset object", {
-    expect_true(class(res) == "NanoStringGeoMxSet")
-    expect_true(nrow(phenoData(res)) == nrow(phenoData(demoData)))
-    expect_true(nrow(featureData(res)) == nrow(featureData(demoData)))
-    # 1 The function outputs a GeoMx S4 class with length same as length of ROIs, sizefact, in phenoData.
-    expect_true("sizefact" %in% varLabels(res))
-    expect_false(any(is.na(res[["sizefact"]])))
-    # 2 The function outputs a GeoMx S4 class with length same as length of negative probes, featfact, in featureData. The value is NA for non-negative probes
-    expect_true(all(is.na(fData(res)$featfact[fData(res)$featfact == "Endogenous"])))
-    expect_identical(!is.na(fData(res)$featfact), fData(res)$CodeClass == "Negative")
+  expect_true(class(res) == "NanoStringGeoMxSet")
+  expect_true(nrow(phenoData(res)) == nrow(phenoData(demoData)))
+  expect_true(nrow(featureData(res)) == nrow(featureData(demoData)))
+  # 1 The function outputs a GeoMx S4 class with length same as length of ROIs, sizefact, in phenoData.
+  expect_true("sizefact" %in% varLabels(res))
+  expect_false(any(is.na(res[["sizefact"]])))
+  # 2 The function outputs a GeoMx S4 class with length same as length of negative probes, featfact, in featureData. The value is NA for non-negative probes
+  expect_true(all(is.na(fData(res)$featfact[fData(res)$featfact == "Endogenous"])))
+  expect_identical(!is.na(fData(res)$featfact), fData(res)$CodeClass == "Negative")
+})
+test_that("fitPoisBG functions with no errors with dgCMatrix",{
+  
+  data("demoData")
+  negdat <- demoData[which(Biobase::fData(demoData)$CodeClass == "Negative"), ]
+  countmat <- Biobase::exprs(negdat)
+  countmat = as(countmat, "dgCMatrix")
+  result <- fitPoisBG(
+    object = countmat,
+    iterations = 10,
+    tol = 1e-3,
+    size_scale = "first")
+  demoData[["sizefact"]] <- result$sizefact[Biobase::sampleNames(demoData)]
+  Biobase::fData(demoData)[["featfact"]] <- NA
+  Biobase::fData(demoData)[["featfact"]][match(names(result$featfact), Biobase::featureNames(demoData), nomatch = 0)] <- result$featfact
+  demoData <- aggreprobe(demoData, use = "cor")
+  expect_true(class(demoData) == "NanoStringGeoMxSet")
+  expect_true(nrow(phenoData(demoData)) == nrow(phenoData(demoData)))
+  expect_true(nrow(featureData(demoData)) == nrow(featureData(demoData)))
+  # 1 The function outputs a GeoMx S4 class with length same as length of ROIs, sizefact, in phenoData.
```

Comment:
Update comment to match what you are testing. Technically we do not confirm length is the same here.

Reply:
Noted but required to ensure the function accepts and processes the right inputs